tests/common/ixia/ixia_fixtures.py

```diff
@@ -310,6 +311,86 @@ def __gen_mac(id):
     """
     return '00:11:22:33:44:{:02d}'.format(id)
 
+def __valid_ipv4_addr(ip):
+    """
+    Determine if a input string is a valid IPv4 address
+
+    Args:
+        ip (unicode str): input IP address
+
+    Returns:
+        True if the input is a valid IPv4 adress or False otherwise
+    """
+    try:
+        return True if type(ip_address(ip)) is IPv4Address else False
+    except ValueError:
+        return False
+
+def __l3_intf_config(config, port_config_list, duthost, ixia_ports):
+    """
+    Generate Tgen configuration of layer 3 interfaces
+
+    Args:
+        config (obj): Tgen API config of the testbed
+        port_config_list (list): list of IXIA port configuration information
+        duthost (object): device under test
+        ixia_ports (list): list of IXIA port information
+
+    Returns:
+        True if we successfully generate configuration or False
+    """
+    mg_facts = duthost.minigraph_facts(host=duthost.hostname)['ansible_facts']
+    l3_intf_facts = mg_facts['minigraph_interfaces']
```

Comment:
We need to check if l3_intf_facts is empty and in that case, return True so that code does not run into issues on T0 topology